plugin/http/httptrace/httptrace.go

```diff
@@ -59,10 +63,75 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := t.base().RoundTrip(req)
 
 	// TODO(jbd): Add status and attributes.
-	trace.EndSpan(ctx)
+	if err != nil {
+		trace.EndSpan(ctx)
+		return resp, err
+	}
+
+	// trace.EndSpan(ctx) will be invoked after
+	// resp.Body.Close() has been invoked.
+	resp.Body = &spanEndBody{rc: resp.Body, spanCtx: ctx}
 	return resp, err
 }
 
+// spanEndBody wraps a response.Body and invokes
+// trace.EndSpan on encountering io.EOF on reading
+// the body of the original response.
+type spanEndBody struct {
+	rc      io.ReadCloser
+	spanCtx context.Context
+
+	endSpanOnce sync.Once
+}
+
+var _ io.ReadCloser = (*spanEndBody)(nil)
+
+func (bpr *spanEndBody) Read(b []byte) (int, error) {
+	n, err := bpr.rc.Read(b)
+	if err == nil {
+		return n, nil
+	}
+
+	// Otherwise, time to end the span or set the status
+	switch err {
+	case io.EOF:
+		bpr.endSpan()
+	default:
+		// For all other errors, set the span status
+		// If the error has a status and code
+		// let's propagate those in the span
+		if ge, ok := err.(*googleapi.Error); ok {
```

Comment:
I agree that status is not required here.

Reply:
@ramonza that PR code is outdated, I replaced it with a code 2 set for any non-nil error. Are y'all asking that I remove any status sets and just process a read as it would normally be done?
Don't worry about the back and forth, it allows for exchange of ideas, so please feel free to propose as many changes ;)

Reply:
@ramonza and @rakyll the difference here is that if we get non2XX(*http.Response.Code) the user can handle that which is the case that you are talking about.
Whereas by the time we are in a *http.Response.Body.Read, we already have a 2XX code, the span is being recorded but we encountered an unexpected error hence we are setting the status of the span and I believe that this case is mutually exclusive of the case where we are setting by HTTP status.